src/app/modules/payments/stripe.controller.ts

```diff
@@ -91,84 +68,157 @@ export const _handleStripeEventUpdates: ControllerHandler<
       meta: {
         action: 'handleStripeEventUpdates',
         req: req.body,
         error: e,
       },
     })
-    return res.status(StatusCodes.BAD_REQUEST).send()
+    return res.sendStatus(StatusCodes.BAD_REQUEST)
+  }
+
+  // Step 2: Received event, proceed to process it.
+
+  const logMeta = {
+    action: 'handleStripeEventUpdates',
+    event,
   }
 
   logger.info({
     message: 'Received Stripe event from webhook',
-    meta: {
-      action: 'handleStripeEventUpdates',
-      event,
-    },
+    meta: logMeta,
   })
 
+  let result: Result<void, any> = ok(undefined)
+
   switch (event.type) {
-    case 'charge.captured':
-    case 'charge.expired':
+    // We catch all payment_intent, charge and payout events, except the
+    // following ignored event types (as associated features are not supported):
+    // - charge.captured, charge.expired (only for capture flow)
+    // - charge.updated (descriptions or metadata are updated)
+    // - payment_intent.amount_capturable_updated (only for capture flow)
+    // - payment_intent.partially_funded (only occurs when payment intents are
+    //   completed in part by customer stripe account balance)
+    case 'payment_intent.canceled':
+    case 'payment_intent.created':
+    case 'payment_intent.payment_failed':
+    case 'payment_intent.processing':
+    case 'payment_intent.requires_action':
+    case 'payment_intent.succeeded':
     case 'charge.failed':
     case 'charge.pending':
     case 'charge.refunded':
-    case 'charge.succeeded':
-    case 'charge.updated':
-      await findPaymentAndUpdate(
+    case 'charge.succeeded': {
+      result = await getMetadataPaymentId(
         event.data.object.metadata,
-        {
-          chargeIdLatest: event.data.object.id,
-          status: stripeChargeStatusToPaymentStatus(event.data.object.status),
-        },
-        event,
+      ).asyncAndThen((paymentId) =>
+        StripeService.processStripeEvent(paymentId, event),
       )
       break
+    }
     case 'charge.dispute.closed':
     case 'charge.dispute.created':
     case 'charge.dispute.funds_reinstated':
     case 'charge.dispute.funds_withdrawn':
-    case 'charge.dispute.updated':
-      await findPaymentAndUpdate(event.data.object.metadata, {}, event)
-      break
-    case 'charge.refund.updated':
-      // We should actually handle this case, need to implement based on get by charge Id though, not doing for hackathon.
+    case 'charge.dispute.updated': {
+      const chargeIdLatest =
+        typeof event.data.object.charge === 'string'
+          ? event.data.object.charge
+          : event.data.object.charge.id
+
+      const payment = await PaymentModel.findOne({ chargeIdLatest })
+      if (!payment) {
+        logger.warn({
+          message: 'Received dispute event with unknown latest charge id',
+          meta: { ...logMeta, chargeIdLatest },
+        })
+        return res.sendStatus(StatusCodes.BAD_REQUEST)
```

Comment:
Hm, this will result in stripe continuing to send us such request 🤔. 

Does it make sense for us to drop it? I assume that subsequent webhook from stripe will still send us nothing, and hog our server.

Reply:
In the situation where we have not responded with a non-`OK`, over three days, stripe might disable our webhook.

Reply:
Understand. I think we don't have enough data to justify either way -- on whether it is risky to keep rejecting or whether there's an unknown bug/race condition.

Agree to keep status quo.